### ISSUE TITLE
Fail to build from sources with ld --as-needed

### ISSUE THREAD

**Issue by GoogleCodeExporter** [chrsmithdemos/leveldb#57]
```
leveldb fails to build from sources with the linking flag '--as-needed' enabled.

The attached patch solves the issue (and set the correct GCC linking order).
```

---

Original issue reported on code.google.com by `quadris...@gmail.com` on 29 Nov 2011 at 11:33

Attachments:
- [0101-as_needed.patch](https://storage.googleapis.com/google-code-attachments/leveldb/issue-57/comment-0/0101-as_needed.patch)

**Comment by GoogleCodeExporter** [comment]
```
The bugref was wrong, attaching the revised patch.
```

---

Original comment by `quadris...@gmail.com` on 29 Nov 2011 at 11:36

Attachments:
- [0101-as_needed.patch](https://storage.googleapis.com/google-code-attachments/leveldb/issue-57/comment-1/0101-as_needed.patch)

**Comment by GoogleCodeExporter** [comment]
```
I've rewritten the patch against the latest snapshot: 36a5f8e
```

---

Original comment by `quadris...@gmail.com` on 5 Dec 2011 at 5:17

Attachments:
- [0101-as_needed.patch](https://storage.googleapis.com/google-code-attachments/leveldb/issue-57/comment-2/0101-as_needed.patch)

**Comment by GoogleCodeExporter** [comment]
```
Again, refreshed patch is attached.
```

---

Original comment by `quadris...@gmail.com` on 13 Apr 2012 at 12:14

Attachments:
- [0101-as_needed.patch](https://storage.googleapis.com/google-code-attachments/leveldb/issue-57/comment-3/0101-as_needed.patch)

**Comment by GoogleCodeExporter** [comment]
```
Forgot to add $(LIBS) to the linking flags to the shared library.
The attached patch fixes it.
```

---

Original comment by `quadris...@gmail.com` on 13 Apr 2012 at 9:47

Attachments:
- [0101-as_needed.patch](https://storage.googleapis.com/google-code-attachments/leveldb/issue-57/comment-4/0101-as_needed.patch)

**Comment by GoogleCodeExporter** [comment]
```
Any news on this bug?
```

---

Original comment by `quadris...@gmail.com` on 13 Oct 2012 at 4:17

**Comment by GoogleCodeExporter** [comment]
```
Sorry, this got dropped on the floor.

Could you take a moment to e-sign the contributor license agreement
at http://code.google.com/legal/individual-cla-v1.0.html
so that I can incorporate your patches.  Email me when you have done so.

Thanks very much.
sanjay@google.com


```

---

Original comment by `san...@google.com` on 16 Oct 2012 at 4:45

**Comment by GoogleCodeExporter** [comment]
```
Fixed in 1.7.0.

```

---

Original comment by `san...@google.com` on 16 Oct 2012 at 11:35
- Changed state: **Fixed**
